File: spec/java/src/io/kaitai/struct/spec/TestInstanceStd.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.*;
import org.testng.annotations.Test;

import java.nio.charset.Charset;
import java.util.ArrayList;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertEqualsNoOrder;

public class TestInstanceStd extends CommonSpec {
    @Test
    public void testInstanceStd() throws Exception {
        InstanceStd r = InstanceStd.fromFile(SRC_DIR + "str_encodings.bin");

        assertEquals(r.header(), "Some ");
    }

}

File: spec/lua/test_instance_std.lua
local luaunit = require("luaunit")

require("instance_std")

TestInstanceStd = {}

function TestInstanceStd:test_instance_std()
    local r = InstanceStd:from_file("src/str_encodings.bin")

    luaunit.assertEquals(r.header, "Some ")
end
